### PR TITLE
Branch enis

### PR DESCRIPTION
**Pull request by sargun**
### Description of the Change

**FIXME**: please add at least one sentence explaining why changes here are being made.

### CODE REVIEW

File: vpc/bpf2/filter.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	__u32		vlan_tci;
+	__be32		addr;
+};
+
+struct bpf_elf_map __section_maps ipv4_map = {
+        .type           = BPF_MAP_TYPE_HASH,
+        .size_key       = sizeof(struct ipv4_key),
+        .size_value     = sizeof(__u16),
+        .pinning        = PIN_GLOBAL_NS,
+        .max_elem       = 10240,
```
Comment (rgulewich) [inline vpc/bpf2/filter.c]: Why 10240?

Comment (sargun) [reply vpc/bpf2/filter.c]: It leaves enough room for leaks, that we wouldn't notice for a while.

File: vpc/bpf2/filter.c [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+	/* TODO: Verify that the dst mac is the mac returned from bpf_fib_lookup
+	   to prevent anyone trying to bypass hairpin routing.
+	 */
+	switch (eth->h_proto) {
+	case __constant_htons(ETH_P_IP):
+		return classifier_egress_filter_v4(skb);
+	case __constant_htons(ETH_P_IPV6):
+		return classifier_egress_filter_v6(skb);
+	case __constant_htons(ETH_P_ARP):
```
Comment (rgulewich) [inline vpc/bpf2/filter.c]: How _does_ ARP work with branch ENIs?

File: vpc/service/assign_address.go [hunk 3]
```diff
@@ -466,3 +469,596 @@ func (vpcService *vpcService) assignAddresses(ctx context.Context, session *ec2w
...
+
+	span.AddAttributes(trace.StringAttribute("currentSecurityGroups", fmt.Sprint(hasSecurityGroupsSet.List())))
+	span.AddAttributes(trace.StringAttribute("newSecurityGroups", fmt.Sprint(wantedSecurityGroupsSet.List())))
+	logger.G(ctx).
+		WithField("currentSecurityGroups", hasSecurityGroupsSet.List()).
+		WithField("newSecurityGroups", wantedSecurityGroupsSet.List()).
+		Info("Changing security groups")
+
+	if !wantedSecurityGroupsSet.Equal(hasSecurityGroupsSet) {
+		if allowSecurityGroupChange {
```
Comment (rgulewich) [inline vpc/service/assign_address.go]: `! allowSecurityGroupChange`?

Comment (sargun) [reply vpc/service/assign_address.go]: Whoops, I forgot to update the branch.

File: vpc/bpf2/filter.c (hunk 2 above)
Comment (sargun) [reply vpc/bpf2/filter.c]: The same way it works in any 802.1q environment. The arp packets are sent over the branch ENI with the vlan id.

**Pull request merged**